moveit_core/collision_distance_field/CMakeLists.txt

```diff
@@ -1,10 +1,5 @@
 set(MOVEIT_LIB_NAME moveit_collision_distance_field)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
Why are we removing this. It is only a warning. Rather than remove, we should mark the overloaded functions as virtual and the overloading functions as override